### PR TITLE
Волхов Михаил 2538

### PR DESCRIPTION
**Pull request by volhovm**
Bitmap, arraycopy, parallelism (i hope so). Number of threads calculating and drawing middle part "stripes" can be edited in STRIPE_THREADS_NUMBER variable.

### CODE REVIEW

**Comment by AVBelyy** [comment]
Sorry for your loss :(

**Comment by volhovm** [comment]
One little optimisation -> ton of bugs suddenly appear -> fix the last one at 23:57:40 -> WHERE IS THE PULL REQUEST BUTTON

**Comment by AVBelyy** [comment]
Hope that @thevery will show you some mercy :)
You are not even the last one to post your code :)

**Comment by nikitacometa** [comment]
Dat button... I hate it so much.

**Comment by volhovm** [comment]
Last version seems to be fast enough to satisfy the requirements. It gives me ~17 FPS (HTC One V) before cache cycling and about 130 after. I hope it will work faster on Nexus because of parallel model used in calculating (my phone has 1 core). DDMS finds no needless allocations except Strings for logs and says that the program spends about 85% of time in StripeDrawer.run() method. Please reply if it's possible to optimize it somewhere I didn't see the possibility.

**Comment by volhovm** [comment]
Ok, it's the last update I hope. 
I've reached ~55 FPS 10 colours without caching (it's not enabled in apks too), ~30 FPS 20 colours.
I will appreciate a lot if this homework is rated on eight-point scale.

File: gradle/local.properties [hunk 1]
```diff
@@ -0,0 +1,11 @@
+## This file is automatically generated by Android Studio.
```
Comment (thevery) [inline gradle/local.properties:1]: should not be commited

File: app/src/main/java/ru/ifmo/md/lesson1/MyActivity.java [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 import android.app.Activity;
 import android.os.Bundle;
+import android.view.MotionEvent;
```
Comment (thevery) [inline app/src/main/java/ru/ifmo/md/lesson1/MyActivity.java:5]: useless change

File: app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java [hunk 3]
```diff
@@ -1,112 +1,307 @@
...
 class WhirlView extends SurfaceView implements Runnable {
-    int [][] field = null;
-    int width = 0;
-    int height = 0;
-    int scale = 4;
-    final int MAX_COLOR = 10;
-    int[] palette = {0xFFFF0000, 0xFF800000, 0xFF808000, 0xFF008000, 0xFF00FF00, 0xFF008080, 0xFF0000FF, 0xFF000080, 0xFF800080, 0xFFFFFFFF};
-    SurfaceHolder holder;
-    Thread thread = null;
+    private final int width = 240;
```
Comment (thevery) [inline app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java:22]: `public static final`

File: app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java [hunk 4]
```diff
@@ -1,112 +1,307 @@
...
+    private final int height = 382;
```
Comment (thevery) [inline app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java:23]: why not 320?

File: app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java [hunk 5]
```diff
@@ -1,112 +1,307 @@
...
+
+    private float k1;
+    private float k2;
```
Comment (thevery) [inline app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java:26]: bad naming

**Comment by thevery** [comment]
ok = 8
~45 FPS at 320*240 as of de438dab72497b1aa158f789a6d6c1f8759fc587

**Comment by volhovm** [comment]
Thanks!

**Pull request closed without merging**